crates/db_schema/src/impls/history_status.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+  source::history_status::{HistoryStatus, HistoryStatusInsertForm, HistoryStatusUpdateForm},
+  traits::Crud,
+  utils::{get_conn, DbPool},
+};
+use diesel::{insert_into, QueryDsl};
+use diesel_async::{AsyncPgConnection, RunQueryDsl};
+use lemmy_db_schema_file::schema::history_status;
+use lemmy_utils::error::{LemmyErrorExt, LemmyErrorType, LemmyResult};
+
+impl Crud for HistoryStatus {
+  type InsertForm = HistoryStatusInsertForm;
+  type UpdateForm = HistoryStatusUpdateForm;
+  type IdType = (String, String);
+
+  async fn create(pool: &mut DbPool<'_>, form: &Self::InsertForm) -> LemmyResult<Self> {
+    let conn = &mut get_conn(pool).await?;
+    insert_into(history_status::table)
+      .values(form)
+      .get_result::<Self>(conn)
+      .await
+      .with_lemmy_type(LemmyErrorType::NotFound)
+  }
+
+  async fn update(
+    pool: &mut DbPool<'_>,
+    id: Self::IdType,
+    form: &Self::UpdateForm,
+  ) -> LemmyResult<Self> {
+    let conn = &mut get_conn(pool).await?;
+    Self::update_conn(conn, id, form).await
+  }
+}
+
+impl HistoryStatus {
+  pub(crate) async fn update_conn(
+    conn: &mut AsyncPgConnection,
+    id: (String, String),
```

Comment:
Change to `id: (&str, &str)` and then convert inside the function body to make call sites simpler.

Reply:
Also theres no need to have `update` and `update_conn` doing the same thing. Remove one of them or remove the `impl Crud`.